File: Swift-Solutions/Problem-07.swift
import Foundation

func isPrime(num: Int) -> Bool {
  if num < 2 {
    return false
  }
  if num == 2 {
    return true
  }
  let limit: Int = Int(pow(Double(num),0.5))+1
  for i in 2...limit {
    if num % i == 0 {
      return false
    }
  }
  return true
}

func nthPrime(n: Int) -> Int {
  var primeCount: Int = 1
  var i: Int = 1
  while primeCount < n {
    i += 2
    if isPrime(num: i) {
      primeCount += 1
    }
  }
  return i
}

print(nthPrime(n: 10_001))

File: C++ Solutions/Problem-07.cpp
/* What is the 10 001st prime number? */

#include <iostream>
using namespace std;

bool isPrime(double num);
int nthPrime(int n);

int main()
{
  int n;
  cout << "Which Prime do yout want to know?\n";
  cin >> n;
  int answer = nthPrime(n);
  cout << "The " + to_string(n) + "th Prime is " + to_string(answer);
  return 0;
}

bool isPrime(double num) {
  if (num <= 1) {
    return false;
  }
  if (num == 2) {
    return true;
  }
  for (int i=2;i<int(sqrt(num))+1;i++) {
    if (int(num) % i == 0) {
      return false;
    }
  }
  return true;
}

int nthPrime(int n) {
  int primes = 1;
  int counter = 1;
  while (primes < n) {
    counter += 2;
    primes += isPrime(counter);
  }
  return counter;
}